snuba/cli/perf.py

```diff
@@ -30,15 +30,10 @@
               default=False, help='Whether or not to profile processing.')
 @click.option('--profile-write/--no-profile-write',
               default=False, help='Whether or not to profile writing.')
-@click.option('--clickhouse-host', default=settings.CLICKHOUSE_HOST,
```

Comment:
What is the plan for other scripts like clenaup and optimize that still use this field and still depend on clickhouse? Are you planning to provide an abstraction on the database (like the writer) for those as well?

Reply:
I don't have an actual fully-formed plan, so take this with a grain of salt, but here's my general thoughts in no particular order or structure:

- Things like the ClickHouse settings should really be considered implementation details of the dataset itself (I think even the fact that the storage engine for the dataset is ClickHouse should be considered an implementation detail, to be honest, to avoid proliferation of using the consumer directly like we have now), so I think there should be some sort of object returned by a method on the dataset object that conforms to an abstract interface (a `CleanupManager` or something, similar to the abstract writer or some of the cdc components) that is responsible for actually executing these tasks. I think this means the simple answer to your question is "yes, eventually"
- For onprem, it'd be ideal if the thing operators had to do was set up a cron for `cleanup` and `optimize` that by default runs for all datasets, and runs the appropriate command on all (storage) hosts in those datasets. I think this will be reasonably straightforward because most onprem installs will probably use a small (potentially single node) cluster that handles both query and storage duties
- This will be slightly more challenging for our use case, since we have query notes (which don't require cleanup and are visible to the application at the moment) and storage nodes (which do require cleanup and are invisible to the application at the moment.) We'll need to find a way to accommodate both of these use cases without making the trivial onprem use case unnecessarily complicated, which will take some thinking

Reply:
ok